packages/react-jss/src/createHoc.js

```diff
@@ -245,31 +228,31 @@ export default function createHOC<
     }
 
     render() {
-      const {dynamicSheet, classes, staticSheet} = this.state
+      const {classes} = this.state
       const {
         innerRef,
         theme,
         jssContext,
         // $FlowFixMe: Flow complains for no reason...
         ...props
       } = this.props
-      const sheet = dynamicSheet || staticSheet
 
-      if (injectMap.sheet && !props.sheet && sheet) props.sheet = sheet
-      if (injectMap.theme) props.theme = theme
+      if (injectTheme && props.theme) props.theme = theme
```

Comment:
I am not sure we even need injectTheme prop

Reply:
Always inject theme?

Reply:
Actually always injecting theme shouldn't be a big problem, the theme shouldn't change often that it would cause a lot of rerenders

Reply:
I would say yes, btw that line doesn't make sense, bc you have no theme in props, props is the rest ... so we seem to lack a test for this

Reply:
We have a test for that, the tests are failing anyway right now